### PR TITLE
added colalteraization ratio

### PR DESCRIPTION
**Pull request by nathan111100**
Added stats for collateralization ratio, incentive difference, and incentive ratio

### CODE REVIEW

File: experiments/dutch_auctions/dutch_auctions_stats.py [hunk 1]
```diff
@@ -46,8 +46,65 @@ def track_stat(state, action, results):
...
+            sale = clip.sales[sale_id]
+            incentive = clip.tip + sale.tab * Rad(clip.chip)
+            state["stats"]["net_diff"] -= incentive
+        elif action["key"] == "TAKE":
+            state["stats"]["net_diff"] += results[0]
+    return track_stat
+
+
+#differenceratio
+def net_diff():
```
Comment (akirillo) [inline experiments/dutch_auctions/dutch_auctions_stats.py:78]: change to `difference_ratio`

File: experiments/dutch_auctions/dutch_auctions_stats.py [hunk 2]
```diff
@@ -46,8 +46,65 @@ def track_stat(state, action, results):
...
+    def track_stat(state, results, action):
```
Comment (akirillo) [inline experiments/dutch_auctions/dutch_auctions_stats.py:79]: have this stat return the ratio of cumulative incentives paid to cumulative bids placed.
sum into `state["stats"]["difference_ratio"]["cumulative_incentive"]` and `state["stats"]["difference_ratio"]["cumulative_bid"]` every timestep, and save the ratio of the two in `state["stats"]["difference_ratio"]["ratio"]`

File: experiments/dutch_auctions/dutch_auctions_stats.py [hunk 3]
```diff
@@ -46,8 +46,65 @@ def track_stat(state, action, results):
...
+    return track_stat
+
+#colateralization ratio
+def colf():
+    def track_stat(state, results, action):
+        num = 0
+        denom = 0
+        if action["key"] == "TAKE":
+
+            for i in results[1]:
```
Comment (akirillo) [inline experiments/dutch_auctions/dutch_auctions_stats.py:106]: will be cleaner to access from `state["vat"]["urns"]` here

File: experiments/dutch_auctions/dutch_auctions_stats.py [hunk 4]
```diff
@@ -46,8 +46,65 @@ def track_stat(state, action, results):
...
 def auction_debt():
-    def track_stat(state, action, _results):
+    def track_stat(state, action):
         if action["key"] == "T_START":
             state["stats"]["auction_debt"] = state["dog"].Dirt
 
     return track_stat
+#net difference
+def net_diff():
+    def track_stat(state, results, action):
```
Comment (akirillo) [inline experiments/dutch_auctions/dutch_auctions_stats.py:56]: let's have this be cumulative, just remove the if case for `"T_START"`

File: experiments/dutch_auctions/dutch_auctions_stats.py [hunk 5]
```diff
@@ -46,8 +46,65 @@ def track_stat(state, action, results):
 
 
 def auction_debt():
-    def track_stat(state, action, _results):
```
Comment (akirillo) [inline experiments/dutch_auctions/dutch_auctions_stats.py:49]: undo lol

File: experiments/dutch_auctions/dutch_auctions_stats.py (hunk 1 above)
Comment (nathan111100) [reply experiments/dutch_auctions/dutch_auctions_stats.py]: done

File: experiments/dutch_auctions/dutch_auctions_stats.py (hunk 2 above)
Comment (nathan111100) [reply experiments/dutch_auctions/dutch_auctions_stats.py:94]: done

File: experiments/dutch_auctions/dutch_auctions_stats.py (hunk 3 above)
Comment (nathan111100) [reply experiments/dutch_auctions/dutch_auctions_stats.py]: done

File: experiments/dutch_auctions/dutch_auctions_stats.py (hunk 4 above)
Comment (nathan111100) [reply experiments/dutch_auctions/dutch_auctions_stats.py:56]: ok, to still account for initialization, I did  if action["key"] == "T_START" and "difference_ratio" not in state["stats"]:
such that everything is still initialied and initialization happens once

File: experiments/dutch_auctions/dutch_auctions_stats.py (hunk 5 above)
Comment (nathan111100) [reply experiments/dutch_auctions/dutch_auctions_stats.py:49]: ok

**Pull request closed without merging**